From 02dca60b881c6cff588c84d4f7ec323deb1ce0b9 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Mon, 3 Jun 2024 12:03:29 +0200
Subject: [PATCH] arm64: dts: st: rework exti node for upstream on stm32mp211

Due to upstream review, drop the interrupt-map child node and use
the interrupts-extended property.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I9ae2e1d80a15733832e0798a3979b19568611b7c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/384707
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 240 ++++++++++++++-----------
 1 file changed, 136 insertions(+), 104 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -1823,70 +1823,88 @@
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
 			#interrupt-cells = <2>;
-			#address-cells = <0>;
 			reg = <0x44220000 0x400>;
-			exti-interrupt-map {
-				#address-cells = <0>;
-				#interrupt-cells = <2>;
-				interrupt-map-mask = <0xffffffff 0>;
-				interrupt-map =
-					<0  0 &intc 0 0 GIC_SPI 268 IRQ_TYPE_LEVEL_HIGH>,
-					<1  0 &intc 0 0 GIC_SPI 269 IRQ_TYPE_LEVEL_HIGH>,
-					<2  0 &intc 0 0 GIC_SPI 270 IRQ_TYPE_LEVEL_HIGH>,
-					<3  0 &intc 0 0 GIC_SPI 271 IRQ_TYPE_LEVEL_HIGH>,
-					<4  0 &intc 0 0 GIC_SPI 272 IRQ_TYPE_LEVEL_HIGH>,
-					<5  0 &intc 0 0 GIC_SPI 273 IRQ_TYPE_LEVEL_HIGH>,
-					<6  0 &intc 0 0 GIC_SPI 274 IRQ_TYPE_LEVEL_HIGH>,
-					<7  0 &intc 0 0 GIC_SPI 275 IRQ_TYPE_LEVEL_HIGH>,
-					<8  0 &intc 0 0 GIC_SPI 276 IRQ_TYPE_LEVEL_HIGH>,
-					<9  0 &intc 0 0 GIC_SPI 277 IRQ_TYPE_LEVEL_HIGH>,
-					<10 0 &intc 0 0 GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>,
-					<11 0 &intc 0 0 GIC_SPI 279 IRQ_TYPE_LEVEL_HIGH>,
-					<12 0 &intc 0 0 GIC_SPI 280 IRQ_TYPE_LEVEL_HIGH>,
-					<13 0 &intc 0 0 GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>,
-					<14 0 &intc 0 0 GIC_SPI 282 IRQ_TYPE_LEVEL_HIGH>,
-					<15 0 &intc 0 0 GIC_SPI 283 IRQ_TYPE_LEVEL_HIGH>,
-					<16 0 &intc 0 0 GIC_SPI 0   IRQ_TYPE_LEVEL_HIGH>,
-					<17 0 &intc 0 0 GIC_SPI 1   IRQ_TYPE_LEVEL_HIGH>,
-					<18 0 &intc 0 0 GIC_SPI 211 IRQ_TYPE_LEVEL_HIGH>,
-					<19 0 &intc 0 0 GIC_SPI 210 IRQ_TYPE_LEVEL_HIGH>,
-					<21 0 &intc 0 0 GIC_SPI 95  IRQ_TYPE_LEVEL_HIGH>,
-					<22 0 &intc 0 0 GIC_SPI 97  IRQ_TYPE_LEVEL_HIGH>,
-					<23 0 &intc 0 0 GIC_SPI 124 IRQ_TYPE_LEVEL_HIGH>,
-					<26 0 &intc 0 0 GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>,
-					<27 0 &intc 0 0 GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>,
-					<28 0 &intc 0 0 GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>,
-					<29 0 &intc 0 0 GIC_SPI 123 IRQ_TYPE_LEVEL_HIGH>,
-					<30 0 &intc 0 0 GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>,
-					<31 0 &intc 0 0 GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>,
-					<32 0 &intc 0 0 GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>,
-					<36 0 &intc 0 0 GIC_SPI 99  IRQ_TYPE_LEVEL_HIGH>,
-					<37 0 &intc 0 0 GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>,
-					<38 0 &intc 0 0 GIC_SPI 112 IRQ_TYPE_LEVEL_HIGH>,
-					<39 0 &intc 0 0 GIC_SPI 137 IRQ_TYPE_LEVEL_HIGH>,
-					<40 0 &intc 0 0 GIC_SPI 138 IRQ_TYPE_LEVEL_HIGH>,
-					<41 0 &intc 0 0 GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>,
-					<43 0 &intc 0 0 GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>,
-					<44 0 &intc 0 0 GIC_SPI 158 IRQ_TYPE_LEVEL_HIGH>,
-					<47 0 &intc 0 0 GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>,
-					<48 0 &intc 0 0 GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>,
-					<59 0 &intc 0 0 GIC_SPI 152 IRQ_TYPE_LEVEL_HIGH>,
-					//	<59 0 &intc 0 0 GIC_SPI 153 IRQ_TYPE_LEVEL_HIGH>,
-					<61 0 &intc 0 0 GIC_SPI 154 IRQ_TYPE_LEVEL_HIGH>,
-					//	<61 0 &intc 0 0 GIC_SPI 155 IRQ_TYPE_LEVEL_HIGH>,
-					<64 0 &intc 0 0 GIC_SPI 185 IRQ_TYPE_LEVEL_HIGH>,
-					<67 0 &intc 0 0 GIC_SPI 9   IRQ_TYPE_LEVEL_HIGH>,
-					<68 0 &intc 0 0 GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>,
-					<70 0 &intc 0 0 GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>,
-					<72 0 &intc 0 0 GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>,
-					<73 0 &intc 0 0 GIC_SPI 172 IRQ_TYPE_LEVEL_HIGH>,
-					<74 0 &intc 0 0 GIC_SPI 96  IRQ_TYPE_LEVEL_HIGH>,
-					<75 0 &intc 0 0 GIC_SPI 98  IRQ_TYPE_LEVEL_HIGH>,
-					<76 0 &intc 0 0 GIC_SPI 125 IRQ_TYPE_LEVEL_HIGH>,
-					<77 0 &intc 0 0 GIC_SPI 205 IRQ_TYPE_LEVEL_HIGH>,
-					<78 0 &intc 0 0 GIC_SPI 206 IRQ_TYPE_LEVEL_HIGH>,
-					<79 0 &intc 0 0 GIC_SPI 207 IRQ_TYPE_LEVEL_HIGH>;
-			};
+			interrupts-extended =
+				<&intc GIC_SPI 268 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_0 */
+				<&intc GIC_SPI 269 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 270 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 271 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 272 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 273 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 274 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 275 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 276 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 277 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_10 */
+				<&intc GIC_SPI 279 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 280 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 282 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 283 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 0   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 1   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 211 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 210 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,						/* EXTI_20 */
+				<&intc GIC_SPI 95  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 97  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 124 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 123 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_30 */
+				<&intc GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 99  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 112 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 137 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 138 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_40 */
+				<&intc GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<&intc GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 158 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,						/* EXTI_50 */
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 152 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,						/* EXTI_60 */
+				<&intc GIC_SPI 154 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 185 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 9   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 118 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<&intc GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_70 */
+				<0>,
+				<&intc GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 172 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 96  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 98  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 125 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 205 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 206 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 207 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
 		syscfg: syscon@44230000 {
@@ -2004,48 +2022,62 @@
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
 			#interrupt-cells = <2>;
-			#address-cells = <0>;
 			reg = <0x442d0000 0x400>;
-
-			exti-interrupt-map {
-				#address-cells = <0>;
-				#interrupt-cells = <2>;
-				interrupt-map-mask = <0xffffffff 0>;
-				interrupt-map =
-					<0  0 &intc 0 0 GIC_SPI 15  IRQ_TYPE_LEVEL_HIGH>,
-					<1  0 &intc 0 0 GIC_SPI 16  IRQ_TYPE_LEVEL_HIGH>,
-					<2  0 &intc 0 0 GIC_SPI 17  IRQ_TYPE_LEVEL_HIGH>,
-					<3  0 &intc 0 0 GIC_SPI 18  IRQ_TYPE_LEVEL_HIGH>,
-					<4  0 &intc 0 0 GIC_SPI 19  IRQ_TYPE_LEVEL_HIGH>,
-					<5  0 &intc 0 0 GIC_SPI 20  IRQ_TYPE_LEVEL_HIGH>,
-					<6  0 &intc 0 0 GIC_SPI 21  IRQ_TYPE_LEVEL_HIGH>,
-					<7  0 &intc 0 0 GIC_SPI 22  IRQ_TYPE_LEVEL_HIGH>,
-					<8  0 &intc 0 0 GIC_SPI 23  IRQ_TYPE_LEVEL_HIGH>,
-					<9  0 &intc 0 0 GIC_SPI 24  IRQ_TYPE_LEVEL_HIGH>,
-					<10 0 &intc 0 0 GIC_SPI 25  IRQ_TYPE_LEVEL_HIGH>,
-					<11 0 &intc 0 0 GIC_SPI 26  IRQ_TYPE_LEVEL_HIGH>,
-					<12 0 &intc 0 0 GIC_SPI 27  IRQ_TYPE_LEVEL_HIGH>,
-					<13 0 &intc 0 0 GIC_SPI 28  IRQ_TYPE_LEVEL_HIGH>,
-					<14 0 &intc 0 0 GIC_SPI 29  IRQ_TYPE_LEVEL_HIGH>,
-					<15 0 &intc 0 0 GIC_SPI 30  IRQ_TYPE_LEVEL_HIGH>,
-					<16 0 &intc 0 0 GIC_SPI 10  IRQ_TYPE_LEVEL_HIGH>,
-					<17 0 &intc 0 0 GIC_SPI 11  IRQ_TYPE_LEVEL_HIGH>,
-					<21 0 &intc 0 0 GIC_SPI 12  IRQ_TYPE_LEVEL_HIGH>,
-					<22 0 &intc 0 0 GIC_SPI 13  IRQ_TYPE_LEVEL_HIGH>,
-					<26 0 &intc 0 0 GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>,
-					<29 0 &intc 0 0 GIC_SPI 181 IRQ_TYPE_LEVEL_HIGH>,
-					<30 0 &intc 0 0 GIC_SPI 182 IRQ_TYPE_LEVEL_HIGH>,
-					<31 0 &intc 0 0 GIC_SPI 183 IRQ_TYPE_LEVEL_HIGH>,
-					<35 0 &intc 0 0 GIC_SPI 5   IRQ_TYPE_LEVEL_HIGH>,
-					<36 0 &intc 0 0 GIC_SPI 4   IRQ_TYPE_LEVEL_HIGH>,
-					<37 0 &intc 0 0 GIC_SPI 6   IRQ_TYPE_LEVEL_HIGH>,
-					<39 0 &intc 0 0 GIC_SPI 2   IRQ_TYPE_LEVEL_HIGH>,
-					<40 0 &intc 0 0 GIC_SPI 3   IRQ_TYPE_LEVEL_HIGH>,
-					<49 0 &intc 0 0 GIC_SPI 201 IRQ_TYPE_LEVEL_HIGH>,
-					<51 0 &intc 0 0 GIC_SPI 202 IRQ_TYPE_LEVEL_HIGH>,
-					<52 0 &intc 0 0 GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH>,
-					<53 0 &intc 0 0 GIC_SPI 204 IRQ_TYPE_LEVEL_HIGH>;
-			};
+			interrupts-extended =
+				<&intc GIC_SPI 15  IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_0 */
+				<&intc GIC_SPI 16  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 17  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 18  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 19  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 20  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 21  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 22  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 23  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 24  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 25  IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_10 */
+				<&intc GIC_SPI 26  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 27  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 28  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 29  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 30  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 10  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 11  IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,						/* EXTI_20 */
+				<&intc GIC_SPI 12  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 13  IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 181 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 182 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_30 */
+				<&intc GIC_SPI 183 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 5   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 4   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 6   IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<&intc GIC_SPI 2   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 3   IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_40 */
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 201 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,						/* EXTI_50 */
+				<&intc GIC_SPI 202 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 204 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
 		rtc: rtc@46000000 {
